src/lib/components/Swap/Input.tsx

```diff
@@ -81,11 +82,22 @@ export default function Input({ disabled, focused }: InputProps) {
     return insufficientBalance ? 'error' : undefined
   }, [balance, inputCurrencyAmount, swapInputCurrencyAmount])
 
+  const amount = useInputOutputFieldAmount(
+    useMemo(
```

Comment:
This does not need to be memoized because it is destructured by the hook and used in a useMemo within the hook, so the references should be the same.